pyscriptjs/src/pyodide.ts

```diff
@@ -91,20 +91,15 @@ export class PyodideRuntime extends Runtime {
 
     async loadFromFile(path: string): Promise<void> {
         const filename = getLastPath(path);
-        await this.run(
-            `
-                from pyodide.http import pyfetch
-                from js import console
-
-                try:
-                    response = await pyfetch("${path}")
-                except Exception as err:
-                    console.warn("PyScript: Access to local files (using 'paths:' in py-config) is not available when directly opening a HTML file; you must use a webserver to serve the additional files. See https://github.com/pyscript/pyscript/issues/257#issuecomment-1119595062 on starting a simple webserver with Python.")
-                    raise(err)
-                content = await response.bytes()
-                with open("${filename}", "wb") as f:
-                    f.write(content)
-            `,
-        );
+        try {
+            const buffer = await fetch(path).then(res => res.arrayBuffer());
+            const data = new Uint8Array(buffer);
+            const stream = this.interpreter.FS.open(filename, 'w');
+            this.interpreter.FS.write(stream, data, 0, data.length, 0);
+            this.interpreter.FS.close(stream);
+        }
+        catch (err) {
+            console.warn("PyScript: Access to local files (using 'paths:' in py-config) is not available when directly opening a HTML file; you must use a webserver to serve the additional files. See https://github.com/pyscript/pyscript/issues/257#issuecomment-1119595062 on starting a simple webserver with Python.")
+        }
```

Comment:
I know that this is a problem which was already present before, but this looks like a good occasion to improve it.
There are many things which could go wrong in the `try` clause, and this message is not always the most appropriate one (e.g., if you make a typo in your `paths=[...]` you would get this, probably).

I think that:

1. if we cannot fetch a file, we should abort the rest of the execution. Errors should not pass silently.

2. corollary of the above, if we abort the execution we should show a big red message on the PAGE, not on the console. I expect that many pyscript beginners don't even know how to look at the console.

3. The "access to local files" message should be shown only if we know that we are trying to access a local file (e.g. by looking at the protocol of the URL and/or `window.location.href`).

Reply:
@antocuni I guess the function `handleFetchError` at https://github.com/pyscript/pyscript/blob/main/pyscriptjs/src/utils.ts#L67 already handles this at one level above.

Reply:
actually, I think we can and should remove `try...catch` blocks from here since `loadFromFile` is itself under a `try...catch` inside `fetchPaths`

```
            try {
                await runtime.loadFromFile(singleFile);
            } catch (e) {
                //Should we still export full error contents to console?
                handleFetchError(<Error>e, singleFile);
            }
```

and `handleFetchError` covers the cases you talked about...

Reply:
done